projects/igniteui-angular/src/lib/grids/selection/selection.service.ts

```diff
@@ -528,17 +529,11 @@ export class IgxGridSelectionService {
         this.emitRowSelectionEvent(newSelection, added, [], event);
     }
 
-    public areAllRowSelected(): boolean {
-        if (!this.grid.data) {
-            return false;
-        }
-        if (this.allRowsSelected !== undefined) {
-            return this.allRowsSelected;
-        }
-
-        const dataItemsID = this.getRowIDs(this.allData);
-        return this.allRowsSelected = Math.min(this.rowSelection.size, dataItemsID.length) > 0 &&
-            new Set(Array.from(this.rowSelection.values()).concat(dataItemsID)).size === this.rowSelection.size;
+    public areAllRowSelected(newSelection?): boolean {
+        const selectedData = newSelection ? newSelection : [...this.rowSelection]
+        const allData = this.getRowIDs(this.allData);
+        const unSelectedRows = allData.filter(row => !selectedData.includes(row));
+        return this.allRowsSelected = this.allData.length > 0 && unSelectedRows.length === 0;
```

Comment:
I don't see no place where `this.allRowsSelected` is read. It is only being set couple of times. Seems we should go without it at all.